contracts/ContinuousTokenFactory.sol

```diff
@@ -2,9 +2,164 @@
 
 pragma solidity >=0.6.0 <0.7.0;
 
+import "@openzeppelin/contracts/math/SafeMath.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
+import "./ERC20Managed.sol";
+import "./Curve.sol";
 
 contract ContinuousTokenFactory {
+  using SafeMath for uint256;
 
+  ERC20 public reserveToken;
+
+  struct ContinuousToken {
+    ERC20Managed token;
+    Curve curve;
+  }
+
+  mapping(bytes32 => ContinuousToken) public continuousTokens;
+
+  constructor(address _reserveToken) public {
+    require(
+      _reserveToken != address(0),
+      "ContinuousTokenFactory: no reserve token ERC20 address"
+    );
+    reserveToken = ERC20(_reserveToken);
+  }
+
+  event NewContinuousToken(
+    bytes32 id,
+    address token,
+    string name,
+    address curve
+  );
+
+  function newContinuousToken(
+    bytes32 id, 
+    string memory name, 
+    string memory symbol,
+    address _curve
+  ) public returns (bool) {
+    require(!exists(id), "ContinuousTokenFactory: continuous token already exists");
+    require(_curve != address(0), "ContinuousTokenFactory: curve at address 0");
+    // TODO find a way to require a Curve contract be at curve
+
+    ERC20Managed token = new ERC20Managed(name, symbol);
+    Curve curve = Curve(_curve);
+    ContinuousToken memory continuousToken = ContinuousToken(token, curve);
+    continuousTokens[id] = continuousToken;
+
+    emit NewContinuousToken(
+      id,
+      address(token),
+      name,
+      address(curve)
+    );
+    return true;
+  }
+
+  event Buy(
+    bytes32 id, 
+    string name,
+    uint256 amount,
+    uint256 price,
+    address buyer,
+    address owner
```

Comment:
I get why we need this, but I'm thinking the names should be `buyer->from` and `owner->to` if we need both?